scripts/uosc/elements/Menu.lua

```diff
@@ -150,6 +158,8 @@ function Menu:update(data)
 		local parent_id = menu.parent_menu and not menu.parent_menu.is_root and menu.parent_menu.id
 		if not menu.is_root then
 			menu.id = (parent_id and parent_id .. ' > ' or '') .. (menu_data.title or i)
+		else
+			menu.id = 'main'
```

Comment:
Add `id` to `MenuData` type so that menu implementers can define their own. Use it as 1st `if` statement here when defined:

```lua
if menu_data.id then
  menu.id = menu_data.id
elseif ...
```